src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { FilterTypes } from '../../types/FilterTypes';
+import { FilterTypesForm } from '../../types/FilterTypesForm';
+import { UserInfo } from '../../types/UserInfo';
+
+import styles from './Filter.module.scss';
+
+export const Filter: React.FC<FilterTypes> = ({ list, setIsFilter }) => {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<FilterTypesForm>({
+    mode: 'onSubmit',
+  });
+
+  const onSubmit = (data: FilterTypesForm) => {
+    if (data.firstName) {
+      let filter = list.filter((item: UserInfo) => {
+        return item.firstName.toLocaleLowerCase().includes(data.firstName.toLocaleLowerCase());
+      });
+      setIsFilter(filter);
+    }
+
+    if (data.gender) {
+      let filter = list.filter((item: UserInfo) => {
+        return item.gender === data.gender;
+      });
+      setIsFilter(filter);
+    }
+
+    if (data.ageFrom && data.ageTo) {
+      let filter = list.filter((item: UserInfo) => {
+        return item.age > data.ageFrom && item.age < data.ageTo;
+      });
+      setIsFilter(filter);
+    }
```

Comment:
we should be able to filter the list with different options
eg age > 5 && gender = male && firstName includes "b"